tdigest.go

```diff
@@ -22,10 +25,17 @@ import (
 //
 // MergeInto(other) will add all of the data within a TDigest into
 // other, combining them into one larger TDigest.
+//
+// Finally, TDigests implement the encoding package's BinaryMarshaler and
```

Comment:
Changing the interface by adding these methods is a breaking change—if other implementations don't happen to have these new methods, their builds will fail.

Reply:
Right, good point. I've taken this opportunity to clean up the API and return a struct. This will be v2.0.0.